opensearchpy/connection/http_urllib3.py

```diff
@@ -214,9 +214,14 @@ def __init__(
         if pool_maxsize and isinstance(pool_maxsize, int):
             kw["maxsize"] = pool_maxsize
 
-        self.pool = pool_class(
-            self.hostname, port=self.port, timeout=self.timeout, **kw
-        )
+        def urllib3_pool_factory() -> Any:
```

Comment:
Why is this not at the class level? Is it because of `**kw`?

Reply:
Yes, do you having any better suggestion?

Reply:
Use a `lambda`?

Reply:
That triggers a warning in the linter because the lambda is not type annotated

Reply:
You can suppress it with a comment?

Reply:
done